File: simulation_prototype/Product.py
from ProductInterface import ProductInterface


class Product(ProductInterface):
    def __init__(self, new_id, new_name, new_price, new_prepareTime):
        super().__init__()

        self._ID = new_id
        self._name = new_name
        self._price = new_price
        self._prepareTime = new_prepareTime


    def getID(self):
        return self._ID


    def getName(self):
        return self._name


    def getPrice(self):
        return self._price


    def getPrepareTime(self):
        return self._prepareTime

File: Product.cpp
#include "Product.h"

#include <iostream>

Product::Product(int new_id, std::string new_name, int new_price, int new_prepareTime)
{
    ID = new_id;
    name = new_name;
    price = new_price;
    prepareTime = new_prepareTime;
}

int Product::getID()
{
    return ID;
}
std::string Product::getName()
{
    return name;
}
int Product::getPrice()
{
    return price;
}
int Product::getPrepareTime()
{
    return prepareTime;
}

std::ostream& operator<<(std::ostream& os, const Product& product)
{
    os << product.ID << " | " << product.name << " | " << product.price << std::endl;
    return os;
}